addons/plugins/auth0_plugin.py

```diff
@@ -0,0 +1,153 @@
+from typing import Tuple, Optional
+
+import jwt
+import requests
+from auth0.v3.exceptions import TokenValidationError
+from django.contrib.auth import get_user_model
+from django.core.handlers.wsgi import WSGIRequest
+from django.core.exceptions import ValidationError
+from saleor.plugins.base_plugin import BasePlugin, ExternalAccessTokens
+
+
+from auth0.v3.authentication.token_verifier import (
+    TokenVerifier,
+    AsymmetricSignatureVerifier,
+)
+
+from django.conf import settings
+
+User = get_user_model()
+
+JWKS_URL = f"https://{settings.AUTH0_DOMAIN}/.well-known/jwks.json"
+ISSUER = f"https://{settings.AUTH0_DOMAIN}/"
+
+
+def verify_token(token):
+    sv = AsymmetricSignatureVerifier(JWKS_URL)
+    tv = TokenVerifier(
+        signature_verifier=sv, issuer=ISSUER, audience=settings.AUTH0_CLIENT_ID
+    )
+
+    try:
+        tv.verify(token)
+    except TokenValidationError as e:
+        raise ValidationError(f"Invalid token! {e}")
+
+
+def get_user_from_token(token) -> Optional[User]:
+    user_id_token = jwt.decode(token, options={"verify_signature": False})
+    user_email = user_id_token["email"]
+    email_verified = user_id_token["email_verified"]
+
+    if not email_verified:
+        raise ValidationError("User email hasn'\t been verified yet.")
+
+    try:
+        user = User.objects.get(email=user_email)
+    except User.DoesNotExist:
+        raise ValidationError(f"Unknown user {user_email}")
+
+    return user
+
+
+class Auth0Plugin(BasePlugin):
+    PLUGIN_NAME = "Authenticate with Auth0"
+    PLUGIN_ID = "vonq.authentication.auth0"
+    PLUGIN_DESCRIPTION = ""
+    CONFIG_STRUCTURE = None
+    DEFAULT_ACTIVE = True
+
+    def external_authentication_url(
+        self, data: dict, request: WSGIRequest, previous_value
+    ) -> dict:
+
+        return {
+            "authorizeUrl": f"https://{settings.AUTH0_DOMAIN}/authorize?"
+            f"response_type=code&"
+            f"client_id={settings.AUTH0_CLIENT_ID}&"
+            f"redirect_uri=https://localhost:9000/callback&"
+            f"scope=openid+profile+email+offline_access&"
+            f"state={data.get('state')}"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `get` method without a default argument to return the value of a key in a dictionary. We recommended that you use a default argument so that if the value for your key is not found, a default value is returned. If a default value is not provided and the key is not found, then `None` is returned.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)

Reply:
Nice! Can we have CodeGuru everywhere? 💯

Reply:
sure! but unfortunately it only supports Python and Java

Reply:
Can you login to https://sonarcloud.io/ using your github? I'll add you to a trial org that I created, with a demo of pkb and mapi analysed.